enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,495 @@
+This enhancement is about a high-level API which will satisfy the needs of most
+customers with strong regulatory or security requirement, while being easy to configure for them and feasible to 
+maintain in the future.
+
+For customers with more advanced requirements we propose a profile that audit logs nearly everything (with exception of security-critical resources), 
+and the customer can filter the output on his own. 
+
+### Goals
+
+- add an abstract audit policy configuration to `apiservers.config.openshift.io/v1`
+  - with few predefined policies
+  - easily extensible in the future 
+  - few knobs on the level of concrete OpenShift customer requirement
+- applied by kube-apiserver, openshift-apiserver and oauth-apiserver.
+
+### Non-Goals
+
+- configuration of audit backends like webhooks
+- configuration of file logger parameters
+
+## Proposal
+
+We propose to add a `audit.profile` field to `apiservers.config.openshift.io/v1` which selects the 
+audit policy to be deployed to all OpenShift-provided APIservers in the cluster. From the beginning
+we provide the following profile:
+
+- `Default`: this is the existing default policy. `Default` is also the default value in the `APIServer` custom resource:
+
+  ```yaml
+    apiVersion: audit.k8s.io/v1beta1
+    kind: Policy
+    # Don't generate audit events for all requests in RequestReceived stage.
+    omitStages:
+    - "RequestReceived"
+    rules:
+    # Don't log requests for events
+    - level: None
+      resources:
+      - group: ""
+        resources: ["events"]
+    # Don't log oauth tokens as metadata.name is the secret
+    - level: None
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+    # Don't log authenticated requests to certain non-resource URL paths.
+    - level: None
+      userGroups: ["system:authenticated", "system:unauthenticated"]
+      nonResourceURLs:
+      - "/api*" # Wildcard matching.
+      - "/version"
+      - "/healthz"
+      - "/readyz"
+  
+    # Here the tail begins which differs per profile:
+  
+    # A catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - "RequestReceived"
+   ```
+
+- `OauthUserRequests`: this is like `Default`, but it logs request bodies for requests for oauth users only, i.e. by users outside of the cluster.
+- `NonOpenShiftRequests`: this is like `Default`, but it logs request bodies for requests not coming from known OpenShift components like kubelets, controllers and operators.
+- `AllRequests`: this logs all request bodies (with the exception of the following).
+
+All of the profiles have in common that security sensitive resources, namely
+
+- `secrets`
+- `routes.route.openshift.io`
```

Comment:
add `oauthclients.oauth.openshift.io` (due to their `.secret`) field

Reply:
done